diff --git a/external/qt/CMakeLists.txt b/external/qt/CMakeLists.txt
index 577b67b..6754d23 100644
--- a/external/qt/CMakeLists.txt
+++ b/external/qt/CMakeLists.txt
@@ -215,8 +215,9 @@ if (WIN32)
             D3D12
             SetupAPI
             Shcore
-            Synchronization
-            WindowsApp
+            Secur32
+            WinHttp
+            ncrypt
         )
     else()
         target_link_libraries(external_qt
diff --git a/options_win.cmake b/options_win.cmake
index c2d66cf..ea978c2 100644
--- a/options_win.cmake
+++ b/options_win.cmake
@@ -52,6 +52,7 @@ if (MSVC)
         /wd4702 # unreachable code
         /wd4310 # cast truncates constant value
         /wd4127 # conditional expression is constant
+        /wd4189 # local variable is initialized but not referenced (MSVC 14.50+)
 
         # Taken from Qt 6.
         # https://developercommunity.visualstudio.com/content/problem/139261/msvc-incorrectly-defines-cplusplus.html
@@ -138,6 +139,7 @@ INTERFACE
     Dwmapi
     UxTheme
     Wtsapi32
+    Comctl32
     Crypt32
     Propsys
     Bcrypt
